src/kernel/task.zig

```diff
@@ -99,6 +104,25 @@ pub const Task = struct {
         return task;
     }
 
+    pub fn createFromElf(program_elf: elf.Elf, kernel: bool, task_vmm: *vmm.VirtualMemoryManager(arch.VmmPayload), allocator: *Allocator) (bitmap.Bitmap(usize).BitmapError || vmm.VmmError || elf.Error || Allocator.Error)!*Task {
+        const task = try create(program_elf.header.entry_address, kernel, task_vmm, allocator);
+        errdefer task.destroy(allocator);
+
+        // Iterate over sections
+        for (program_elf.section_headers) |header, i| {
```

Comment:
If there are multiple headers, lets say 4, and the in the 3rd loop the vaddr allocation fails, then there will still be dangling addresses that will need to be freed. So will need a keep track of all the vaddrs and free then if there is a failure. Could also add a test for this.

Reply:
Well spotted